src/internal/channel.js

```diff
@@ -124,16 +124,27 @@ export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
   }
 
   const chan = channel(buffer)
+  const close = () => {
+    if(!chan.__closed__) {
+      if (unsubscribe) {
+        unsubscribe()
+      }
+      chan.close()
```

Comment:
ahh, so we typically unsubscribe before closing the channel. thank you for the fix @Andarist 👍